server/controllers/rpaFrameworkCommandsController.js

```diff
@@ -1,25 +1,29 @@
-const activityDataRetrieval = require('../services/ActivityDataRetrieval/ActivityDataRetrieval');
+const mongoose = require('mongoose');
+const rpaModels = require('../models/rpaTaskModel');
 
+// GET /rpa-framework/commands/get-available-applications
 exports.getAvailableApplications = async (req, res) => {
   try {
-    const listOfDistinctApplications = await activityDataRetrieval.getDistinctApplicationsFromDB();
     res.set('Content-Type', 'application/json');
-    res.send(listOfDistinctApplications);
+    mongoose.connect(process.env.MONGODB_URI, { useNewUrlParser: true, useUnifiedTopology: true });    
+    mongoose.model('rpa-tasks').distinct('Application', (err, tasks) => {
```

Comment:
Same goes for the `.model` call. You are already creating the model object in `models/rpaTaskModel.js` but are not using it.
Usually, the models are created once from the schemas and then imported when needed. (See for example [this tutorial](https://www.robinwieruch.de/mongodb-express-setup-tutorial))